lib/modules/versioning/nuget/index.ts

```diff
@@ -1,36 +1,362 @@
+  }
+
+  isSingleVersion(version: string): boolean {
+    const r = parseRange(version);
+    if (!r) {
+      return false;
+    }
+
+    return r.type === 'nuget-exact-range';
+  }
+
+  isStable(version: string): boolean {
+    const v = parseVersion(version);
+    if (v) {
+      return !v.prerelease;
+    }
+
+    const r = parseRange(version);
+    if (!r || r.type !== 'nuget-exact-range') {
+      return false;
+    }
+
+    return !r.version.prerelease;
+  }
+
+  isValid(input: string): boolean {
+    const v = parseVersion(input);
+    if (v) {
+      return true;
+    }
+
+    const r = parseRange(input);
+    if (r) {
+      return true;
+    }
+
+    return false;
+  }
+
+  isVersion(input: string | undefined | null): boolean {
+    if (!input) {
+      return false;
+    }
+
+    const v = parseVersion(input);
+    if (!v) {
+      return false;
+    }
+
+    return true;
+  }
+
+  getMajor(version: string): number | null {
+    const v = parseVersion(version);
+    if (!v) {
+      return null;
+    }
+
+    return v.major;
+  }
+
+  getMinor(version: string): number | null {
+    const v = parseVersion(version);
+    if (!v) {
+      return null;
+    }
+
+    return v.minor ?? null;
+  }
+
+  getPatch(version: string): number | null {
+    const v = parseVersion(version);
+    if (!v) {
+      return null;
+    }
+
+    return v.patch ?? null;
+  }
+
+  equals(version: string, other: string): boolean {
+    const x = parseVersion(version);
+    const y = parseVersion(other);
+    if (!x || !y) {
+      return false;
+    }
+
+    return compare(x, y) === 0;
+  }
+
+  isGreaterThan(version: string, other: string): boolean {
+    const x = parseVersion(version);
+    const y = parseVersion(other);
+    if (!x || !y) {
+      return false;
+    }
+
+    return compare(x, y) > 0;
+  }
+
+  isLessThanRange(version: string, range: string): boolean {
+    const v = parseVersion(version);
+    if (!v) {
+      return false;
+    }
+
+    const u = parseVersion(range);
+    if (u) {
+      return compare(v, u) < 0;
+    }
+
+    const r = parseRange(range);
+    if (!r) {
+      return false;
+    }
+
+    if (r.type === 'nuget-exact-range') {
+      return compare(v, r.version) < 0;
+    }
+
+    if (r.type === 'nuget-bracket-range') {
+      if (!r.min) {
+        return false;
+      }
+
+      const minBound =
+        r.min.type === 'nuget-version'
+          ? r.min
+          : getFloatingRangeLowerBound(r.min);
+      const cmp = compare(v, minBound);
+      return r.minInclusive ? cmp < 0 : cmp <= 0;
+    }
+
+    const minBound = getFloatingRangeLowerBound(r);
+    return compare(v, minBound) < 0;
+  }
+
+  getSatisfyingVersion(versions: string[], range: string): string | null {
+    if (!this.isValid(range)) {
+      return null;
+    }
+
+    let max: string | null = null;
+    let maxParsed: NugetVersion | undefined;
+    for (const version of versions) {
+      if (this.matches(version, range)) {
```

Comment:
can we create a util function, which takes the parsed versions, so we don't need to parse version twice and range even more (perf)